src/classes/Contacts.cls

```diff
@@ -0,0 +1,713 @@
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Deletes a Household object when no Contacts are related to it. Called by the AfterDelete
+    * trigger handler on Contact.  Also detects if other Contacts remain in the Household, and thus the
+    * Household needs its names updated.
+    * @param contacts List of Contacts being deleted
+    * @param dmlWrapper On return, holds the Households to delete.
+    * @return void
+    ********************************************************************************************************/
+    public void deleteOrRenameHouseholdsWhenContactDeleted(Contact[] contacts, TDTM_Runnable.DmlWrapper dmlWrapper) {
+
+        list<SObject> householdDeletes = new list<SObject>();
+        map<Id, Integer> householdCountMap = new map<Id, Integer>();
+        list<id> contactsNeedingHouseholdsRenamed = new list<id>();
+
+        //Generate a list of Household Ids
+        for (Contact c : contacts) {
+            if (c.npo02__household__c != null)
+                householdCountMap.put(c.npo02__household__c, 0);
+        }
+
+        //Loop through all Contacts at one of the Household Ids
+        for (Contact c : [Select Id, npo02__Household__c from Contact where npo02__Household__c in :householdCountMap.keyset()]) {
+            Integer lastCount = householdCountMap.get(c.npo02__household__c);
+            lastCount += 1;
+            householdCountMap.put(c.npo02__household__c, lastCount);
+        }
+
+        //Loop through all the contacts to determine if they can be deleted
+        for (Contact c : contacts) {
+            if (householdCountMap.get(c.npo02__household__c) < 1) {
+                //If there are no other Contacts on the Account
+                householdDeletes.add(new npo02__Household__c(Id = c.npo02__household__c));
+            } else {
+                //otherwise, we need to update the household
+                contactsNeedingHouseholdsRenamed.add(c.npo02__household__c);
+            }
+        }
+
+        if (householdDeletes.size() > 0) {
+            dmlWrapper.objectsToDelete.addAll(householdDeletes);
+        }
+
+        if (contactsNeedingHouseholdsRenamed.size() > 0) {
+            updateHouseholdNames(contactsNeedingHouseholdsRenamed);
+        }
+    }
+
+    /*******************************************************************************************************
+     * @description kicks off Opportunity Rollups after contacts have been merged, but only if NOT using
+     * the Household Account model
+     * @param winningContacts Set of Contact Id's
+     * @return void
+     ********************************************************************************************************/
+    public static void contactMergeUpdate(Set<Id> winningContacts) {
+
+        npe01__Contacts_And_Orgs_Settings__c candoSettings = UTIL_CustomSettingsFacade.getContactsSettings();
```

Comment:
lol. I just read the var name as "Can Do Settings". Never noticed that before.  :)